openedx/core/djangoapps/user_authn/cookies.py

```diff
@@ -38,34 +95,24 @@ def standard_cookie_settings(request):
         'httponly': None,
     }
 
+    # In production, TLS should be enabled so that this cookie is encrypted
+    # when we send it.  We also need to set "secure" to True so that the browser
+    # will transmit it only over secure connections.
+    #
+    # In non-production environments (acceptance tests, devstack, and sandboxes),
+    # we still want to set this cookie.  However, we do NOT want to set it to "secure"
+    # because the browser won't send it back to us.  This can cause an infinite redirect
+    # loop in the third-party auth flow, which calls `is_logged_in_cookie_set` to determine
+    # whether it needs to set the cookie or continue to the next pipeline stage.
+    cookie_settings['secure'] = request.is_secure()
```

Comment:
Nit: It feels like it would be safer to make this based on a config setting I imagine we have, but maybe this is good enough since we know this code can only be reached via secure requests in Production.

Reply:
Yup, I think this is good enough.  Also, note that I didn't change this behavior.  Pre-existing cookies were already marked this way before this PR; except for the EDXMKTG_LOGGED_IN_COOKIE_NAME, which I believe can be consistent with the other cookies.